File: find_inflection_point.swift
#!/usr/bin/env swift
// http://programmingpraxis.com/2015/08/21/two-homework-problems/

func inflectionPoint(n: [Int]) -> (Int,Int) {
  var inflectionPoint = (0,1)
  for i in 1..<(n.count-1) {
    if difference(n,i,i+1) < difference(n,inflectionPoint.0,inflectionPoint.1) {
      inflectionPoint = (i,i+1)
    } 
  }
  return (n[inflectionPoint.0],n[inflectionPoint.1])
}

func difference(n: [Int], a: Int, b: Int) -> Int {
  let firstHalf = n[0...a].reduce(0) { $0 + $1 }
  let secondHalf = n[b..<n.count].reduce(0) { $0 + $1 }
  return firstHalf > secondHalf ? firstHalf - secondHalf : secondHalf - firstHalf
}

let testArray = [3, 7, 9, 8, 2, 5, 6]
println(inflectionPoint(testArray))

File: find_inflection_point.js
#!/usr/bin/env node
// http://programmingpraxis.com/2015/08/21/two-homework-problems/

function inflectionPoint(list) {
  var inflectionIndex = 0;
  var infectionAmount = list.reduce(function(a,b) { return a+b });
  for (var i = 1; i < list.length; i++) {
    if (difference(list,i) < infectionAmount) {
      inflectionIndex = i; infectionAmount = difference(list,i);
    }
  }
  return {inflectionIndex:inflectionIndex,infectionAmount:infectionAmount}
}

function difference(list,i) {
  var firstPart = list.slice(0,i).reduce(function(a,b) { return a+b });
  var secondPart = list.slice(i,list.length).reduce(function(a,b) { return a+b });
  return firstPart > secondPart ? firstPart - secondPart : secondPart - firstPart;
}

var t = [3, 7, 9, 8, 2, 5, 6];
console.log(inflectionPoint(t));
